### PR TITLE
Fix for #263: prevent overflow in multipart chunk offset reconstruction

### PR DESCRIPTION
**Pull request by peterhillman**
Minor change. The original test is probably redundant now, but I'm not completely sure that there isn't an edge case

### CODE REVIEW

File: OpenEXR/IlmImf/ImfMultiPartInputFile.cpp [hunk 1]
```diff
@@ -639,13 +639,18 @@ MultiPartInputFile::Data::chunkOffsetReconstruction(OPENEXR_IMF_INTERNAL_NAMESPA
                 int y_coordinate;
                 OPENEXR_IMF_INTERNAL_NAMESPACE::Xdr::read <OPENEXR_IMF_INTERNAL_NAMESPACE::StreamIO> (is, y_coordinate);
                 
+                
+                if(y_coordinate < header.dataWindow().min.y || y_coordinate > header.dataWindow().max.y)
+                {
+                    // bail to exception catcher: y out of range. Test now to prevent overflow in following arithmetic
+                   throw int();
```
Comment (meshula) [inline OpenEXR/IlmImf/ImfMultiPartInputFile.cpp]: throw int() occurs four times in this file, and nowhere else. I suspect it's legacy. Probably 
````
throw IEX_NAMESPACE::IoExc("y out of range");
````
and similar should replace all the throw int() in this file. Should we hit all four in this commit?

Comment (peterhillman) [reply OpenEXR/IlmImf/ImfMultiPartInputFile.cpp]: I'll do that. I think I originally wrote those intending to refactor them out.
I think they can be replaced with `break` but throwing an exception with a message as you suggest will make it easier to debug

**Review by meshula — APPROVED** [review]
looks good, thanks!

File: OpenEXR/IlmImf/ImfMultiPartInputFile.cpp [hunk 2]
```diff
@@ -573,8 +573,7 @@ MultiPartInputFile::Data::chunkOffsetReconstruction(OPENEXR_IMF_INTERNAL_NAMESPA
             
             if(partNumber<0 || partNumber>int(parts.size()))
```
Comment (cary-ilm) [inline OpenEXR/IlmImf/ImfMultiPartInputFile.cpp]: Whenever we encounter T() or (T) casts, we should replace them with static_cast<T>, it just looks more intentional.
